frontend/src/container/MetricsTable/Metrics.test.tsx

```diff
@@ -1,34 +1,47 @@
-import { render, RenderResult, screen, waitFor } from '@testing-library/react';
-import { ReactElement } from 'react';
-import { Provider } from 'react-redux';
+import { render, screen, waitFor } from '@testing-library/react';
 import { BrowserRouter } from 'react-router-dom';
-import {
-	combineReducers,
-	legacy_createStore as createStore,
-	Store,
-} from 'redux';
 
-import { InitialValue } from '../../store/reducers/metric';
 import Metrics from './index';
 
-const rootReducer = combineReducers({
-	metrics: (state = InitialValue) => state,
-});
-
-const mockStore = createStore(rootReducer);
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useLocation: (): { pathname: string } => ({
+		pathname: 'localhost:3001/services/',
+	}),
+}));
 
-const renderWithReduxAndRouter = (mockStore: Store) => (
-	component: ReactElement,
-): RenderResult =>
-	render(
-		<BrowserRouter>
-			<Provider store={mockStore}>{component}</Provider>
-		</BrowserRouter>,
-	);
+const services = [
+	{
+		serviceName: 'frontend',
+		p99: 1261498140,
+		avgDuration: 768497850.9803921,
+		numCalls: 255,
+		callRate: 0.9444444444444444,
+		numErrors: 0,
+		errorRate: 0,
+		num4XX: 0,
+		fourXXRate: 0,
+	},
+	{
+		serviceName: 'customer',
+		p99: 890150740.0000001,
+		avgDuration: 369612035.2941176,
+		numCalls: 255,
+		callRate: 0.9444444444444444,
+		numErrors: 0,
+		errorRate: 0,
+		num4XX: 0,
+		fourXXRate: 0,
+	},
+];
 
 describe('Metrics Component', () => {
 	it('renders without errors', async () => {
-		renderWithReduxAndRouter(mockStore)(<Metrics />);
```

Comment:
can we have the common component wrapper which wrap the Query Provider and Browser Router

Reply:
Creating a component that will wrap QueryProvider and BrowserRouter is fine, but in this unit test case, we are only using BrowserRouter, so for this test case it's better to wrap the component with only BrowserRouter. The QueryProvider and BrowserRouter wrapper component can be part of another PR where is it needed.